### PR TITLE
feat: publish package to registry (DCP-1814) 🎉

### PR DESCRIPTION
**Pull request by vinayakkulkarni**
- [x] Enables ship.js trigger to publish to GPR
- [x] Updated meta files & removed duplicate prettier config files
- [x] Clean up GitHub action workflows
- [x] Add support for Husky & Commitlint to adhere to [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) so we can follow SemVer
- [x] Add implementation in https://github.com/dscvr-one/dscvr-media-worker/pull/76

### CODE REVIEW

**Review by jimmymadrigal — CHANGES_REQUESTED** [review]
Shouldn't we send this to npmjs instead as a private package?

**Comment by vinayakkulkarni** [comment]
> Shouldn't we send this to npmjs instead as a private package?

Since this is a public fork, it is a public package.

File: package.json [hunk 1]
```diff
@@ -1,67 +1,84 @@
 {
-  "name": "link-preview-js",
-  "version": "3.0.5",
-  "description": "Javascript module to extract and fetch HTTP link information from blocks of text.",
-  "main": "build/index.js",
+  "name": "@dscvr-one/link-preview-js",
+  "description": "JavaScript module to extract and fetch HTTP link information from blocks of text.",
+  "version": "0.0.0",
+  "sideEffects": false,
```
Comment (ncpenke) [inline package.json:5]: what does `sideEffects` do?

**Review by ncpenke — APPROVED** [review]
LGTM - but please wait till approval from @jimmymadrigal or @PipDscvr. Thank you!

File: package.json (hunk 1 above)
Comment (vinayakkulkarni) [reply package.json:5]: Basically tells bundlers we aren’t doing any global imports or css imports to cause side effects while tree shaking

good explanation: https://stackoverflow.com/a/49776306

**Review by jimmymadrigal — APPROVED** [review]

**Pull request merged**